<?xml version="1.0"?>
<launch>
    <arg name="world_file" default="$(find my_robot_gazebo)/worlds/empty_track.world" />
    <arg name="paused" default="false" />
    <arg name="robot_name" default="smart" />

    <include file="$(find gazebo_ros)/launch/empty_world.launch">
        <arg name="paused" value="$(arg paused)" />
        <arg name="world_name" value="$(arg world_file)" />
    </include>

    <include file="$(find my_robot_gazebo)/launch/spawn_robot.launch">
        <arg name="robot_name" value="$(arg robot_name)" />
    </include>

    <!-- Broadcast TF from ground truth odometry -->
    <node pkg="my_robot_gazebo" type="ground_truth_tf_broadcaster.py" name="ground_truth_tf" output="screen">
        <param name="odom_topic" value="/$(arg robot_name)/ground_truth/state"/>
        <param name="parent_frame" value="world"/>
        <param name="child_frame" value="base_link"/>
    </node>
</launch>
